src/MessagePack.SourceGenerator/CodeAnalysis/QualifiedNamedTypeName.cs

```diff
@@ -183,14 +183,19 @@ public override string GetQualifiedName(Qualifiers qualifier = Qualifiers.Global
                 case "UInt64": return "ulong";
                 case "UInt32": return "uint";
                 case "UInt16": return "ushort";
+                case "Single": return "float";
+                case "Double": return "double";
+                case "Decimal": return "decimal";
+                case "Boolean": return "bool";
                 case "Byte": return "byte";
                 case "Int64": return "long";
                 case "Int32": return "int";
                 case "Int16": return "short";
                 case "SByte": return "sbyte";
                 case "Char": return "char";
                 case "String": return "string";
-                case "ValueTuple": return $"({string.Join(", ", this.TypeArguments.Select(ta => ta.GetQualifiedName(qualifier, genericStyle, includeNullableAnnotation)))})";
+                case "Object": return "object";
+                case "ValueTuple": return string.Format(this.TypeArguments.Length > 1 ? "({0})" : "global::System.ValueTuple<{0}>", string.Join(", ", this.TypeArguments.Select(ta => ta.GetQualifiedName(qualifier, genericStyle, includeNullableAnnotation))));
```

Comment:
We can simplify this by focusing it on just the scenario that works, and let the rest of the code in the method handle the single type arg case:
```suggestion
                case "ValueTuple" when this.TypeArguments.Length > 1: return $"({string.Join(", ", this.TypeArguments.Select(ta => ta.GetQualifiedName(qualifier, genericStyle, includeNullableAnnotation)))})";
```